feat: added files section to navigate the filesystem in use by the filesystem MCP server

cm94242 opened the pull request:
Related Task: #722 

## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [x] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues #eg101 

## No File Gateway Configured

<img width="910" height="499" alt="image" src="https://github.com/user-attachments/assets/fb02700f-f8ba-4845-b54f-a80055b74715" />

## Filesystem UI

| no content | file system structure + Agent |
|---|---|
| <img width="1164" height="682" alt="image" src="https://github.com/user-attachments/assets/f2ff9266-7594-496f-9f8b-5f9f8ef3cbee" /> | <img width="1368" height="684" alt="image" src="https://github.com/user-attachments/assets/1177d2a1-bbac-46fb-9851-6ab8b5e1cd4b" /> |

## File size limit

<img width="541" height="186" alt="image" src="https://github.com/user-attachments/assets/9793ec54-8f3f-4c0a-9d1d-cfaa5781d4ee" />

cm94242 committed:
files

havasik commented on services/ark-dashboard/ark-dashboard/lib/api/files-client.ts:
```diff
@@ -0,0 +1,7 @@
+import { APIClient } from './client';
+
+export const filesApiClient = new APIClient(
+  process.env.NEXT_PUBLIC_FILES_API_BASE_URL ||
```
This will only work with devspace localhost

daniele-marostica replied:
Updated this to use the `/api/v1/proxy/:serviceName/:path` proxy endpoint exposed by ARK API layer.

- `serviceName` is the hostname of the marketplace service to fetch (in this case [`file-gateway-api` is used as default value](https://github.com/mckinsey/agents-at-scale-marketplace/pull/88/changes#diff-97c8b3494627c6e04ec86c187bd7457431289a8f4994fc6c755728faba2f7119R5))
- `path` is the marketplace endpoint path to fetch (eg: `/files/my-file.txt/download`)

This proxy endpoint has been created to:

- avoid coupling dashboard to marketplace services
- avoid CORS issues on ark dashboard to fetch data from marketplace services running on the same cluster
- avoid hardcoding k8s internal DNS records

danielemarostica committed:
Merge branch 'main' into madden/file-browser

danielemarostica committed:
working on filesystem ui

## ARK API

- added `/proxy/:serviceName` endpoint to proxy requests to another service available in the same cluster

## ARK Dashboard

- added empty screen in case no filesystem MCP is configured to help user configuring one
- replaced hardcoded api calls to file gateway api to the proxy endpoint to avoid CORS issues

danielemarostica committed:
Merge branch 'main' into madden/file-browser

danielemarostica committed:
fixed failing tests

danielemarostica committed:
fixed build

danielemarostica committed:
Merge branch 'main' into madden/file-browser

danielemarostica committed:
added copy path, tests

danielemarostica committed:
persist filesystem navigation in session

danielemarostica committed:
fixed eslint typings

danielemarostica committed:
split proxy api handlers

danielemarostica committed:
fix flaky test

danielemarostica committed:
added link

danielemarostica committed:
limiting file upload to 1MB

added link to RAG documentation

danielemarostica committed:
added option to click to browse files

david-gombos commented on services/ark-dashboard/ark-dashboard/middleware.ts:
```diff
@@ -15,6 +15,8 @@ async function middleware(request: NextRequest) {
   const apiPath = `${basePath}/api/`;
 
   if (request.nextUrl.pathname.startsWith(apiPath)) {
+    console.log('[middleware] Intercepted API request:', request.nextUrl.pathname);
```
What does the added logs help?

dwmkerr-agent replied:
agreed, will delete happy path logging

dwmkerr replied:
Done - removed all console.log statements from middleware.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+    if query_params:
+        target_url += "?" + "&".join(f"{k}={v}" for k, v in query_params.items())
+
+    logger.info(f"Proxying {request.method} request to {target_url}")
```
remove happy path logs

dwmkerr replied:
Done - removed happy path logging.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+                headers=filtered_headers,
+            )
+    except httpx.RequestError as e:
+        logger.error(f"Error proxying request to {target_url}: {e}")
```
confirm; this should not be needed as throwing an exception already logs to stderr

dwmkerr replied:
Confirmed - removed logger.error call and unused logging import.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+        )
+
+
+@router.get("/{service_name}/{api_path:path}")
```
confirm, api docs needed for rich swagger

dwmkerr replied:
Confirmed - docstrings are included for OpenAPI/Swagger documentation.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+
+
+@router.post("/{service_name}/{api_path:path}")
+async def proxy_request_post(
```
same, pls confirm for all routes.

dwmkerr replied:
Confirmed - all routes have docstrings for Swagger.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+
+
+@router.options("/{service_name}/{api_path:path}")
+async def proxy_request_options(
```
confirm - is duplication by verb needed

dwmkerr replied:
Yes, separate decorators per HTTP verb are needed for FastAPI to generate proper OpenAPI documentation with distinct operation IDs. Using `@router.api_route(methods=[...])` would lose the explicit per-method swagger docs.

dwmkerr-agent commented on services/ark-api/ark-api/src/ark_api/api/v1/proxy.py:
```diff
@@ -0,0 +1,134 @@
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/proxy", tags=["proxy"])
```
confirm; must be in api docs (even if high level)

dwmkerr replied:
Confirmed - added to ark-api docs at `docs/content/developer-guide/services/ark-api.mdx` under Service Proxy section.

dwmkerr-agent commented on services/ark-dashboard/ark-dashboard/.env.example:
```diff
@@ -30,4 +30,6 @@ NEXT_PUBLIC_TOKEN_REFRESH_INTERVAL_MS=
-AUTH_MODE=sso
\ No newline at end of file
+AUTH_MODE=sso
+#Base URL for the Ark API (optional, defaults to /api/v1/proxy/file-gateway-api)
```
why is this needed, if for local testing maybe another way

daniele-marostica replied:
this ENV is not needed at all, let's just use the default address where needed

dwmkerr-agent commented on services/ark-dashboard/ark-dashboard/components/app-sidebar.tsx:
```diff
@@ -110,8 +114,19 @@ export function AppSidebar() {
 
+    const checkFilesAPIHealth = async () => {
+      try {
+        await filesService.list({ max_keys: 1 });
```
is listing a file the way we are checking whether the service is present? if so I wonder if there is a better way eg `HEAD /proxy/service/<servicename>` which checks for presence / metadata in k8s? also we could hit `/resources/services/<servicename>` - for discussion

dwmkerr replied:
Addressed - added `GET /v1/proxy/service` endpoint that returns list of available services in namespace. Dashboard now calls `proxyService.isServiceAvailable('file-gateway-api')` to check service presence via K8s service discovery instead of trying to list files.

dwmkerr-agent reviewed (changes requested):
Review comments, @dwmkerr will do next iteration of comments

danielemarostica committed:
code review

dwmkerr committed:
fix: improve proxy service discovery and address review comments

- Change proxy route from /proxy/{name} to /proxy/service/{name}
- Add GET /v1/proxy/service endpoint to list available services
- Dashboard now checks service availability via K8s service discovery
- Remove happy path logging from proxy.py and middleware.ts
- Add proxy documentation to ark-api.mdx
- Add proxyService to dashboard for service availability checks

daniele-marostica reviewed (changes requested):
Let's update the /proxy endpoint to ensure using the plural syntax `/proxy/services/:serviceName`

dwmkerr committed:
refactor: rename proxy/service to proxy/services

- Update API router prefix to /proxy/services (plural)
- Update dashboard files-client and proxy service URLs
- Update documentation
- Remove unused NEXT_PUBLIC_ARK_API_BASE_URL env var

dwmkerr committed:
chore: regenerate dashboard types for proxy/services route

dwmkerr committed:
chore: merge main

dwmkerr committed:
chore: regenerate types after merge

danielemarostica committed:
code review

danielemarostica committed:
updated types
